src/ucs/datastruct/bitmap.h

```diff
@@ -568,6 +592,75 @@ _UCS_BITMAP_DECLARE_TYPE(128)
 _UCS_BITMAP_DECLARE_TYPE(256)
 
 
+/* Helper function to set all bitmap bits to 0 */
+static UCS_F_ALWAYS_INLINE void
+ucs_bitmap_bits_clear(ucs_bitmap_word_t *bits, size_t num_words)
+{
+    memset(bits, 0, num_words * sizeof(ucs_bitmap_word_t));
+}
+
+
+/* Helper function to return work mask of bit at given index */
+static UCS_F_ALWAYS_INLINE ucs_bitmap_word_t
+ucs_bitmap_word_bit_mask(size_t bit_index)
+{
+    return UCS_BIT(bit_index % UCS_BITMAP_BITS_IN_WORD);
+}
+
+
+/* Helper function to return word index of a given bit index */
+static UCS_F_ALWAYS_INLINE size_t ucs_bitmap_word_index(size_t bit_index,
+                                                        size_t num_words)
+{
+    UCS_BITMAP_CHECK_INDEX(bit_index, num_words, <);
+    return bit_index / UCS_BITMAP_BITS_IN_WORD;
+}
+
+
+/* Helper function to retrieve the value of a single bit */
+static UCS_F_ALWAYS_INLINE int
+ucs_bitmap_bits_get(const ucs_bitmap_word_t *bits, size_t num_words,
+                    size_t bit_index)
+{
+    const ucs_bitmap_word_t mask = ucs_bitmap_word_bit_mask(bit_index);
+    return !!(bits[ucs_bitmap_word_index(bit_index, num_words)] & mask);
+}
+
+
+/* Helper function to set the value of a single bit to 1 */
+static UCS_F_ALWAYS_INLINE void
+ucs_bitmap_bits_set(ucs_bitmap_word_t *bits, size_t num_words, size_t bit_index)
+{
+    const ucs_bitmap_word_t mask = ucs_bitmap_word_bit_mask(bit_index);
+    bits[ucs_bitmap_word_index(bit_index, num_words)] |= mask;
+}
+
+
+/* Helper function to set the value of a single bit to 0 */
+static UCS_F_ALWAYS_INLINE void ucs_bitmap_bits_reset(ucs_bitmap_word_t *bits,
+                                                      size_t num_words,
+                                                      size_t bit_index)
+
+{
+    const ucs_bitmap_word_t mask = ucs_bitmap_word_bit_mask(bit_index);
+    bits[ucs_bitmap_word_index(bit_index, num_words)] &= ~mask;
+}
+
+
+/* Helper function to test if all bitmap bits are 0 */
+static UCS_F_ALWAYS_INLINE int
+ucs_bitmap_bits_is_zero(const ucs_bitmap_word_t *bits, size_t num_words)
+{
+    const ucs_bitmap_word_t *word;
```

Comment:
Minor:
In many other places there's `ucs_bitmap_word_t` **bits** and not **word**. Maybe rename for consistency

Reply:
"word" is when it's used to access a single word, and "bits" is when it is used as an array of words
do you think we should use "const ucs_bitmap_word_t *words" for example, instead of "const ucs_bitmap_word_t *bits"?

Reply:
> do you think we should use "const ucs_bitmap_word_t *words" for example, instead of "const ucs_bitmap_word_t *bits"?

Yes, that sounds good. `const ucs_bitmap_word_t *words` is both meaningful & consistent, IMO.